### PR TITLE
chunkify for TS and TSD

### PR DESCRIPTION
**Pull request by RychenerLorenz**

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -3,5 +3,5 @@
   <component name="JavaScriptSettings">
     <option name="languageLevel" value="ES6" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_14" default="false" project-jdk-name="Pipenv (timeatlas)" project-jdk-type="Python SDK" />
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.7 (timeatlas-kzsAXcI_)" project-jdk-type="Python SDK" />
```
Comment (fredmontet) [inline .idea/misc.xml:6]: Try to avoid making PR with IDE related files

File: src/timeatlas/time_series/time_series.py [hunk 2]
```diff
@@ -201,6 +202,18 @@ def prepend(self, ts: 'TimeSeries'):
         new_series = concat([ts.series, self.series])
         return TimeSeries(new_series, self.metadata)
 
+    def chunkify(self, n: int) -> List['TimeSeries']:
+        """
+        Splits TimeSeries into chunks of length n
+
+        :param n: length of chunks
+        :return: List of TimeSeries
```
Comment (fredmontet) [inline src/timeatlas/time_series/time_series.py:210]: Please make Google Docstrings as stated in: https://timeatlas.dev/contributing/coding_style_guide.html

It's possible to set that up in your IDE (IntelliJ or PyCharm) in Preferences > Tools > Python Integrated Tools > Docstrings > Docstrings format, then choose Google.

File: src/timeatlas/time_series_dataset/time_series_dataset.py [hunk 3]
```diff
@@ -120,6 +121,23 @@ def random(self, n: int, seed: int = None, indices: bool = False) -> Any:
         else:
             TimeSeriesDataset(random.sample(population=self.data, k=n))
 
+    def chunkify(self, n: int) -> List['TimeSeriesDataset']:
+        """
+
+        Splits each TimeSeries in the TimeSeriesDataset into chunks of length n
```
Comment (fredmontet) [inline src/timeatlas/time_series_dataset/time_series_dataset.py:127]: Same here with the docstring

File: src/timeatlas/time_series/time_series.py [hunk 4]
```diff
@@ -201,6 +202,18 @@ def prepend(self, ts: 'TimeSeries'):
...
+        """
+
+        ts_chunks = [TimeSeries(v) for n, v in self.series.groupby(np.arange(len(self.series)) // n)]
```
Comment (fredmontet) [inline src/timeatlas/time_series/time_series.py:213]: The metadata object isn't propagated in the chunks, use the TimeSeries constructor second argument to fix that.

**Review by fredmontet — CHANGES_REQUESTED** [review]
The chunkify method is alright except the metadata propagation. I added a specific review comment for that. Otherwise, in addition to the chunkify methods, you added some other files (IDE and binary). This is ok but not recommended to make clean code (see my comments).

File: .idea/misc.xml (hunk 1 above)
Comment (RychenerLorenz) [reply .idea/misc.xml:6]: I'll update the .gitignore accordingly

**Pull request merged**